docker/Dockerfile

```
@@ -20,6 +20,22 @@ ENV LANG C.UTF-8
 # Set correct environment variables.
 ENV HOME /root
 
+RUN add-apt-repository ppa:deadsnakes/ppa
+
+RUN apt-get update
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    python3.11 \
+    python3-pip \
+    python3-dev \
+    && \
+apt-get clean && \
+rm -rf /var/lib/apt/lists/*
+
+RUN pip3 install --upgrade pip
+
+RUN pip install protobuf==3.20.*
```

Why do we need this ppa ? I suggest using python3 pip available on this distro

```suggestion
RUN apt update && apt install -y python3-pip && \
  RUN pip3 install protobuf==3.20.*
```